examples/medplum-task-example-app/app/src/components/task-actions/AddDueDate.tsx

```diff
@@ -0,0 +1,112 @@
+import { Button, Modal } from '@mantine/core';
+import { notifications } from '@mantine/notifications';
+import { getQuestionnaireAnswers, MedplumClient, normalizeErrorString, PatchOperation } from '@medplum/core';
+import { Period, Questionnaire, QuestionnaireResponse, Task, TaskRestriction } from '@medplum/fhirtypes';
+import { QuestionnaireForm, useMedplum } from '@medplum/react';
+import { IconCircleCheck, IconCircleOff } from '@tabler/icons-react';
+import { useState } from 'react';
+
+interface AddDueDateProps {
+  task: Task;
+  onChange: (updatedTask: Task) => void;
+  medplum: MedplumClient;
```

Comment:
You don't need to pass in `medplum` as a prop to all these handlers. 

Medplum ships with custom hook called `useMedplum()`. You can call this at the the top of your Component to get the medplum client. Will simplify your props

Reply:
React Context Providers